CHANGELOG.md

```diff
@@ -37,6 +37,11 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
 ## Unreleased
 
+### Improvements
+
+* #[131](https://github.com/babylonlabs-io/babylon/pull/131) Add new staking
+events
```

Comment:
This is state machine breaking as it affects some Merkle root (IIRC the data root) of the block ([ref](https://github.com/tendermint/tendermint/issues/7471))

Reply:
Hmm have checked the source code of CometBFT, and it seems that `LastResultsHash` is the [hash](https://github.com/cometbft/cometbft/blob/bbb86ced558f05a60c2402283e0cb13d08dc57a2/state/execution.go#L697-L698) of all responses of `DeliverTx`. So if event schmas are changed this will affect the `LastResultsHash`. So this seems to mean it's consensus breaking?

Reply:
hmm this `TxResultsHash` goes to - https://github.com/cometbft/cometbft/blob/v0.38.x/state/store.go#L412
which create some special `types.NewResults` struct, which only cares about 4 fields from whole response: https://github.com/cometbft/cometbft/blob/v0.38.x/types/results.go#L47 (events are ommitted)

Reply:
Right, good find! Events are not deterministic and this won't be consensus breaking then :+1: